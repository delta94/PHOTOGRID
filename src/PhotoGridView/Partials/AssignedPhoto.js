import React, { Component } from 'react';
import { PropTypes } from 'prop-types';
import ReactTooltip from 'react-tooltip';

export class AssignedPhoto extends Component {
  constructor() {
    super();
    
    this.state = {
      photoOver: false
    };

    this.handleMouseEnter = this.handleMouseEnter.bind(this);
    this.handleMouseLeave = this.handleMouseLeave.bind(this);
  }

  handleMouseEnter() {
    this.setState({
      photoOver: true
    });
  }

  handleMouseLeave() {
    this.setState({
      photoOver: false
    });
  }

  render() {
    const { 
      id, 
      title, 
      image,
      handleClickCloseBtn
    } = this.props;
    const { photoOver } = this.state;
    return (
      <div
        className="assigned-product-photo-container"
      >
        <img
          src={image}
          alt={title}
          className={`br__radius-5 ${photoOver ? 'w-100' : 'w-80'}`}
          style={{ cursor: 'pointer' }}
          onMouseEnter={this.handleMouseEnter}
          onMouseLeave={this.handleMouseLeave}
          data-tip={title}
          data-for="test"
        />
        {
          photoOver && 
          <div
            className="assigned-product-cancel-container"
            onMouseEnter={this.handleMouseEnter}
            onMouseLeave={this.handleMouseLeave}
          >
            <span
              className="icon__cross-close color__white"
              onClick={() => handleClickCloseBtn(id)}
            />
          </div>
        }
        <ReactTooltip 
          id="test"
          place="top"
          effect="solid"
        />
      </div>
    );
  }
}

AssignedPhoto.propTypes = {
  id: PropTypes.number,
  title: PropTypes.string,
  image: PropTypes.string,
  handleClickCloseBtn: PropTypes.func
};

export default AssignedPhoto;