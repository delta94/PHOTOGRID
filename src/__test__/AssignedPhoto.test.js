import React from 'react';
import { shallow, mount } from 'enzyme';

import AssignedPhoto from '../PhotoGridView/Partials/AssignedPhoto';

const mockData = {
  products: [
    {
      id: 1,
      title: 'Whisker',
      image: 'https://ae01.alicdn.com/kf/HTB1N8Q9G7yWBuNjy0Fpq6yssXXal/Led-twig-lights-holiday-lights-branches-Christmas-ornaments-HJW.jpg'
    },
    {
      id: 2,
      title: 'Merry',
      image: 'https://ae01.alicdn.com/kf/HTB1Cvrtl8jTBKNjSZFuq6z0HFXaa/Automatic-Toothpas…oothbrush-Holder-with-Cups-No-Nail-Wall-Stand-Shelf-Bathroom-Organizer.jpg'
    },
    {
      id: 3,
      title: 'Apple',
      image: 'https://img.staticbg.com/thumb/large/oaupload/banggood/images/9B/D2/54eafdb3-7f9c-4d56-9522-73c12c2fe2c3.jpg'
    },
    {
      id: 4,
      title: 'Sample',
      image: 'https://ae01.alicdn.com/kf/HTB1KwN_cGmgSKJjSsplq6yICpXai/Gamiss-Spring-Autu…ual-Tshirt-Plus-Size-Halloween-Batwing-Loose-Long-Sleeve-Solid-Color-T.jpg'
    }
  ]
};
const mockhandler = jest.fn();

describe('AssignedPhoto Component', () => {
  const wrapper = mount(
    <AssignedPhoto
      id = {mockData.products[0].id}
      title = {mockData.products[0].id.title}
      image = {mockData.products[0].id.image}
      key = {mockData.products[0].id}
      handleClickCloseBtn = {mockhandler}
    />
  );

  it('allows us to set props for ProductMan', () => {
    expect(wrapper.props().id).toBe(mockData.products[0].id);
    expect(wrapper.props().title).toBe(mockData.products[0].id.title);
    expect(wrapper.props().image).toBe(mockData.products[0].id.image);
  });

  it('click the close button', () => {
    wrapper.setState({
      photoOver: true
    });
    wrapper.find('span').simulate('click');
    expect(mockhandler).toBeCalled();
  });

});

describe('AssignedPhoto SnapShot', () => {
  const wrapper = shallow(
    <AssignedPhoto
      id = {mockData.products[0].id}
      title = {mockData.products[0].id.title}
      image = {mockData.products[0].id.image}
      key = {mockData.products[0].id}
      handleClickCloseBtn = {jest.fn()}
    />
  );

  it('match to snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});