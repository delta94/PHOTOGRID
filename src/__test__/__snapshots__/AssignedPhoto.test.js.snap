// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssignedPhoto SnapShot match to snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AssignedPhoto
    handleClickCloseBtn={[MockFunction]}
    id={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          className="br__radius-5 w-80"
          data-for="test"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        />,
        false,
        <ReactTooltip
          effect="solid"
          id="test"
          insecure={true}
          place="top"
          resizeHide={true}
          wrapper="div"
        />,
      ],
      "className": "assigned-product-photo-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": undefined,
          "className": "br__radius-5 w-80",
          "data-for": "test",
          "data-tip": undefined,
          "onMouseEnter": [Function],
          "onMouseLeave": [Function],
          "src": undefined,
          "style": Object {
            "cursor": "pointer",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "effect": "solid",
          "id": "test",
          "insecure": true,
          "place": "top",
          "resizeHide": true,
          "wrapper": "div",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            className="br__radius-5 w-80"
            data-for="test"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          />,
          false,
          <ReactTooltip
            effect="solid"
            id="test"
            insecure={true}
            place="top"
            resizeHide={true}
            wrapper="div"
          />,
        ],
        "className": "assigned-product-photo-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": undefined,
            "className": "br__radius-5 w-80",
            "data-for": "test",
            "data-tip": undefined,
            "onMouseEnter": [Function],
            "onMouseLeave": [Function],
            "src": undefined,
            "style": Object {
              "cursor": "pointer",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "effect": "solid",
            "id": "test",
            "insecure": true,
            "place": "top",
            "resizeHide": true,
            "wrapper": "div",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
